### PR TITLE
New resource: `azurerm_container_registry_task_schedule`

### PR DESCRIPTION
**Pull request by magodo**
This feature allows `azurerm_container_registry_task` run right after create/update.

Besides, I noticed that both the 2019-06-01-preview and the 2021-08-01-preview ACR SDK uses the same API version for the ACR task (i.e. 2019-06-01-preview), so I just stick on using the 2021-08-01-preview package only.

Fixes: #15095

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
hi @magodo 

Thanks for this PR - apologies for the delayed review here.

This is a bit of a weird resource, chatting through this with @katbyte we believe that this'd make more sense as a separate resource to schedule the task, rather than a feature flag against the task resource itself.

This'd allows this new resource to take a "keepers" section of values for when this should be re-run, which'd allow users to taint this as necessary - WDYT?

Thanks!

**Comment by magodo** [comment]
@tombuildsstuff I'm not sure whether I understand this correctly, while I've separate the scheduling of the task into its own resource `azurerm_container_registry_task_schedule`. Please take a look to see whether it meents your expectation, thank you!

## Test

```shell
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/containers  -run='TestAccContainerRegistryTaskSchedule_basic'
=== RUN   TestAccContainerRegistryTaskSchedule_basic
=== PAUSE TestAccContainerRegistryTaskSchedule_basic
=== CONT  TestAccContainerRegistryTaskSchedule_basic
--- PASS: TestAccContainerRegistryTaskSchedule_basic (336.78s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    336.795s
```

**Comment by magodo** [comment]
@katbyte I've resolved the merge conflicts and passed the test, would you please take another look?

File: website/docs/r/container_registry_task_schedule.html.markdown [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Manages a Container Registry Task Schedule.
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "example" {
```
Comment (katbyte) [inline website/docs/r/container_registry_task_schedule.html.markdown:19]: we should have blank lines here
```suggestion
}

resource "azurerm_resource_group" "example" {
```

File: website/docs/r/container_registry_task_schedule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+subcategory: "Container"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_container_registry_task_schedule"
+description: |-
+  Manages a Container Registry Task Schedule.
+---
+
+# azurerm_container_registry_task_schedule
+
+Manages a Container Registry Task Schedule.
```
Comment (katbyte) [inline website/docs/r/container_registry_task_schedule.html.markdown:11]: what is this resource doing? it takes a single argument and no properties?

Comment (katbyte) [reply website/docs/r/container_registry_task_schedule.html.markdown:11]: where is the schedule defined?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/container_registry_task_schedule.html.markdown (hunk 2 above)
Comment (magodo) [reply website/docs/r/container_registry_task_schedule.html.markdown:11]: @katbyte This resource is an operation-like resource, which "runs" when created. What it does is to schedule a task run for an existing `azurerm_container_registry_task`.

Comment (katbyte) [reply website/docs/r/container_registry_task_schedule.html.markdown:11]: should the resource be named `azurerm_container_registry_task_schedule_run_now` then as there is no "Schedule" and it just triggers a run now?

Comment (magodo) [reply website/docs/r/container_registry_task_schedule.html.markdown]: @katbyte I've made the change. Meanwhile, I've removed the import test and document for this resource as it is an operation-like resource, it is meaningless to import it.

## Test

```shell
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/containers  -run='TestAccContainerRegistryTaskSchedule_'
=== RUN   TestAccContainerRegistryTaskSchedule_basic
=== PAUSE TestAccContainerRegistryTaskSchedule_basic
=== CONT  TestAccContainerRegistryTaskSchedule_basic
--- PASS: TestAccContainerRegistryTaskSchedule_basic (238.69s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    238.705s
```

**Review by katbyte — APPROVED** [review]
LGTM ☁️

**Pull request merged**